api/v1/slurmcluster_types.go

```diff
@@ -68,6 +68,25 @@ type SlurmClusterSpec struct {
 	//
 	// +kubebuilder:validation:Optional
 	Telemetry *Telemetry `json:"telemetry,omitempty"`
+
+	// PartitionConfiguration define partition configuration of slurm worker nodes
+	// https://slurm.schedmd.com/slurm.conf.html#SECTION_PARTITION-CONFIGURATION
+	// +kubebuilder:validation:Optional
+	PartitionConfiguration PartitionConfiguration `json:"partitionConfiguration,omitempty"`
+}
+
+type PartitionConfiguration struct {
+	// ConfigType
+	// +kubebuilder:validation:Enum=default;custom
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="default"
+	ConfigType string `json:"configType,omitempty"`
+	// RawConfig define partition configuration as raw multi string started with PartitionName
+	// Example for custom ConfigType:
+	// PartitionName=low_priority Nodes=worker-[0-15] Default=YES MaxTime=INFINITE State=UP PriorityTier=1
+	// PartitionName=high_priority  Nodes=worker-[10-20] Default=NO MaxTime=INFINITE State=UP PriorityTier=2
+	// +kubebuilder:validation:Optional
+	RawConfig string `json:"rawConfig,omitempty"`
```

Comment:
Maybe it is better to have it as `[]string` instead of multi-line string?

Reply:
It gives:
- You won't need `scanner` anymore
  - As simple as `for range` loop
  - Line processing becomes more clear
- You can just add comment lines between config strings to separate blocks like it's said [here](https://github.com/nebius/soperator/pull/131#discussion_r1806632926)